<a href="https://colab.research.google.com/github/6170chandan/global-terrorism-data-analysis/blob/main/Global_Terrorism_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount("/content/drive/")

Mounted at /content/drive/


In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
file_path = "/content/drive/MyDrive/Global Terrorism Analysis - Chandan Kumar Verma/Copy of Global Terrorism Data.csv"

In [4]:
terror_file = pd.read_csv(file_path,encoding="ISO-8859-1")

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [9]:
terror_file.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [12]:
name_of_columns = (print(terror_file.columns))

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region',
       ...
       'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource', 'INT_LOG',
       'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object', length=135)


Renaming the columns

In [17]:
terror_file.rename(columns={'eventid':'Eventid', 'iyear':'Year', 'imonth':'Month', 'iday':'Day', 
                          'extended':'Extended', 'resolution':'Resolution',  'attacktype1_txt':'Attacktype', 
                          'country_txt':'Country', 'region_txt':'Region', 'provstate': 'Provstate', 
                          'city':'City', 'crit1':'Crit1', 'crit2':'Crit2', 
                          'crit3':'Crit3', 'multiple':'Multiple', 'success':'Success', 'suicide':'Suicide', 
                          'targtype1_txt':'Targtype', 'natlty1_txt':'Natlty1', 'natlty2_txt':'Natlty2', 
                          'natlty3_txt':'Natlty3', 'gname':'Gname', 'gname2':'Gname2', 'gname3':'Gname3', 
                          'guncertain1':'Guncertain1', 'guncertain2':'Guncertain2', 'guncertain3':'Guncertain3', 
                          'claimed':'Claimed', 'weaptype1_txt':'Weaptype', 'weapsubtype1_txt':'Weapsubtype', 
                          'nkill':'Nkill', 'nkillus':'Nkillus', 'nkillter':'Nkillter', 'nwound':'Nwound', 
                          'nwoundus':'Nwoundus', 'nwoundte':'Nwoundter', 'property':'Property', 
                          'propextent_txt':'Propextent', 'propvalue':'Propvalue', 'ishostkid':'Ishostkid', 
                          'nhostkid':'Nhostkid', 'nhostkidus':'Nhostkidus', 'ransom':'Ransom', 
                          'hostkidoutcome':'Hostkidoutcome', 'nreleased':'Nreleased'},inplace=True)

In [18]:
terror_file

Eventid  Year  Month  Day approxdate  Extended Resolution  \
0       197000000001  1970      7    2        NaN         0        NaN   
1       197000000002  1970      0    0        NaN         0        NaN   
2       197001000001  1970      1    0        NaN         0        NaN   
3       197001000002  1970      1    0        NaN         0        NaN   
4       197001000003  1970      1    0        NaN         0        NaN   
...              ...   ...    ...  ...        ...       ...        ...   
181686  201712310022  2017     12   31        NaN         0        NaN   
181687  201712310029  2017     12   31        NaN         0        NaN   
181688  201712310030  2017     12   31        NaN         0        NaN   
181689  201712310031  2017     12   31        NaN         0        NaN   
181690  201712310032  2017     12   31        NaN         0        NaN   

        country             Country  region  ... addnotes  \
0            58  Dominican Republic       2  ...      NaN   
1           130              Mexico       1  ...      NaN   
2           160         Philippines       5  ...      NaN   
3            78              Greece       8  ...      NaN   
4           101               Japan       4  ...      NaN   
...         ...                 ...     ...  ...      ...   
181686      182             Somalia      11  ...      NaN   
181687      200               Syria      10  ...      NaN   
181688      160         Philippines       5  ...      NaN   
181689       92               India       6  ...      NaN   
181690      160         Philippines       5  ...      NaN   

                                                   scite1  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Somalia: Al-Shabaab Militants Attack Army Che...   
181687  "Putin's 'victory' in Syria has turned into a ...   
181688  "Maguindanao clashes trap tribe members," Phil...   
181689  "Trader escapes grenade attack in Imphal," Bus...   
181690  "Security tightened in Cotabato following IED ...   

                                                   scite2  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 2 ...   
181687  "Two Russian soldiers killed at Hmeymim base i...   
181688                                                NaN   
181689                                                NaN   
181690  "Security tightened in Cotabato City," Manila ...   

                                                   scite3  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 1 ...   
181687  "Two Russian servicemen killed in Syria mortar...   
181688                                                NaN   
181689                                                NaN   
181690                                                NaN   

                        dbsource  INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0                           PGIS        0         0        0       0      NaN  
1                           PGIS

As the columns are 135 which is a huge number so I am going to have only 20 columns to analyze the data.

In [19]:
df = terror_file[['Eventid', 'Year','Country', 'Region', 'Provstate', 'City', 'Crit1', 'Crit2', 'Crit3', 
                  'Success', 'Suicide', 'Attacktype', 'Targtype', 'Natlty1','Gname', 'Guncertain1', 
                  'Claimed', 'Weaptype', 'Nkill','Nwound']]

In [21]:
df.shape

(181691, 20)

In [22]:
df.head()

Eventid  Year             Country                       Region  \
0  197000000001  1970  Dominican Republic  Central America & Caribbean   
1  197000000002  1970              Mexico                North America   
2  197001000001  1970         Philippines               Southeast Asia   
3  197001000002  1970              Greece               Western Europe   
4  197001000003  1970               Japan                    East Asia   

  Provstate           City  Crit1  Crit2  Crit3  Success  Suicide  \
0       NaN  Santo Domingo      1      1      1        1        0   
1   Federal    Mexico city      1      1      1        1        0   
2    Tarlac        Unknown      1      1      1        1        0   
3    Attica         Athens      1      1      1        1        0   
4   Fukouka        Fukouka      1      1      1        1        0   

                       Attacktype                     Targtype  \
0                   Assassination  Private Citizens & Property   
1     Hostage Taking (Kidnapping)      Government (Diplomatic)   
2                   Assassination          Journalists & Media   
3               Bombing/Explosion      Government (Diplomatic)   
4  Facility/Infrastructure Attack      Government (Diplomatic)   

              Natlty1                               Gname  Guncertain1  \
0  Dominican Republic                              MANO-D          0.0   
1             Belgium  23rd of September Communist League          0.0   
2       United States                             Unknown          0.0   
3       United States                             Unknown          0.0   
4       United States                             Unknown          0.0   

   Claimed    Weaptype  Nkill  Nwound  
0      NaN     Unknown    1.0     0.0  
1      NaN     Unknown    0.0     0.0  
2      NaN     Unknown    1.0     0.0  
3      NaN  Explosives    NaN     NaN  
4      NaN  Incendiary    NaN     NaN

In [23]:
df.tail()

Eventid  Year      Country                      Region  \
181686  201712310022  2017      Somalia          Sub-Saharan Africa   
181687  201712310029  2017        Syria  Middle East & North Africa   
181688  201712310030  2017  Philippines              Southeast Asia   
181689  201712310031  2017        India                  South Asia   
181690  201712310032  2017  Philippines              Southeast Asia   

              Provstate           City  Crit1  Crit2  Crit3  Success  Suicide  \
181686  Middle Shebelle  Ceelka Geelow      1      1      0        1        0   
181687         Lattakia         Jableh      1      1      0        1        0   
181688      Maguindanao       Kubentog      1      1      1        1        0   
181689          Manipur         Imphal      1      1      1        0        0   
181690      Maguindanao  Cotabato City      1      1      1        0        0   

                            Attacktype                     Targtype  \
181686                   Armed Assault                     Military   
181687               Bombing/Explosion                     Military   
181688  Facility/Infrastructure Attack  Private Citizens & Property   
181689               Bombing/Explosion         Government (General)   
181690               Bombing/Explosion                      Unknown   

            Natlty1                                       Gname  Guncertain1  \
181686      Somalia                                  Al-Shabaab          0.0   
181687       Russia                           Muslim extremists          0.0   
181688  Philippines  Bangsamoro Islamic Freedom Movement (BIFM)          0.0   
181689        India                                     Unknown          0.0   
181690  Philippines                                     Unknown          0.0   

        Claimed    Weaptype  Nkill  Nwound  
181686      1.0    Firearms    1.0     2.0  
181687      0.0  Explosives    2.0     7.0  
181688      0.0  Incendiary    0.0     0.0  
181689      0.0  Explosives    0.0     0.0  
181690      0.0  Explosives    0.0     0.0

**Finding the country with highest number of terrorist attacks.**

---



In [35]:
Highest_terror_attack=print("Country with highest terror:",df["Country"].value_counts().index[0])

Country with highest terror: Iraq


In [43]:
df['Country'].value_counts()

Iraq                   24636
Pakistan               14368
Afghanistan            12731
India                  11960
Colombia                8306
                       ...  
International              1
Wallis and Futuna          1
South Vietnam              1
Andorra                    1
Antigua and Barbuda        1
Name: Country, Length: 205, dtype: int64